src/runner/porter.runner.ts

```diff
@@ -173,6 +175,116 @@ export class PorterRunner extends Runner {
     }
   }
 
+  async portPhoneServiceInsertEvent(manager: EntityManager, batchSize = 1000): Promise<void> {
+    type LastMemberNotesType = {
+      criteria: {
+        id: FindOperator<any>;
+        appId: string;
+      };
+      lastMemberRecord: {
+        lastMemberNoteAnswered?: Date | undefined;
+        lastMemberNoteCalled?: Date | undefined;
+        lastMemberNoteCreated: Date;
+      };
+    };
+
+    type ErrInfoType = {
+      memberNote: { data: MemberNote; errMsg: string } | 'NoError';
+      member: { data: LastMemberNotesType; errMsg: string } | 'NoError';
+    };
+
+    type ErrMesType = {
+      key: string;
+      date: string;
+      info: ErrInfoType | 'No data found' | 'Data format error';
+    };
+
+    type DataType = {
+      memberNotes: MemberNote;
+      lastMemberNotes: LastMemberNotesType;
+    };
+
+    function hasDataProperty(data: DataType): boolean {
+      return Boolean(
+        data.hasOwnProperty('lastMemberNotes') &&
+          data.lastMemberNotes?.hasOwnProperty('criteria') &&
+          data.lastMemberNotes?.hasOwnProperty('lastMemberRecord') &&
+          data.hasOwnProperty('memberNotes'),
+      );
+    }
+
+    const client = this.cacheService.getClient();
+    const pattern = `PhoneService:*`;
+    const errAry: ErrMesType[] = [];
+    let cursor = '0';
+
+    do {
+      const scanResult = await client.scan(cursor, 'MATCH', pattern, 'COUNT', batchSize);
+      cursor = scanResult[0];
+      const keys = scanResult[1];
+
+      for (const key of keys) {
```

Comment:
In this `for (const key of keys)` loop, let's be cautious of potential N+1 problems and consider bulk operations  to manage performance and runner time limits.